### PR TITLE
use automoc only on openms libs, fix asan flag propagation

### LINKED ISSUE #6503: Visibility issues SQLiteCPP
```
CMake Warning (dev) at src/openms/thirdparty/SQLiteCpp/CMakeLists.txt:183 (add_library):
  Policy CMP0063 is not set: Honor visibility properties for all target
  types.  Run "cmake --help-policy CMP0063" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  Target "SQLiteCpp" of type "STATIC_LIBRARY" has the following visibility
  properties set for CXX:

    CXX_VISIBILITY_PRESET

  For compatibility CMake is not honoring them for this target.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at src/openms/thirdparty/SQLiteCpp/sqlite3/CMakeLists.txt:9 (add_library):
  Policy CMP0063 is not set: Honor visibility properties for all target
  types.  Run "cmake --help-policy CMP0063" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  Target "sqlite3" of type "STATIC_LIBRARY" has the following visibility
  properties set for CXX:

    CXX_VISIBILITY_PRESET

  For compatibility CMake is not honoring them for this target.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

### LINKED ISSUE #6504: SQLiteCPP not working in nightlies
`make[2]: *** No rule to make target 'lib/libSQLiteCpp.a', needed by 'lib/libOpenMS.so'.  Stop.`

https://cdash.openms.de/viewBuildError.php?buildid=21596

Repository revision a0e4a04

### LINKED ISSUE #6508: SQliteCPP breaks pyopenms
https://github.com/OpenMS/OpenMS/actions/runs/3540478391/jobs/5943579970

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

fixes #6508 
fixes #6504 
fixes #6503 

TODO:  currently C++ compiler flags seem to be forwarded to C (but ignored). This should probably be fixed

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by timosachsenberg** [comment]
I think CXX_EXE_LINKER_FLAGS doesn't exist and we need something like
```
set_target_properties(abc_proxy_with_asan PROPERTIES COMPILE_FLAGS "-g3 -fsanitize=address -fno-omit-frame-pointer")
set_target_properties(abc_proxy_with_asan PROPERTIES LINK_FLAGS "-fsanitize=address -static-libasan")
```

**Comment by jpfeuffer** [comment]
Yes. But mind that this will not sanitize the tools then.

**Comment by jpfeuffer** [comment]
I think it would be ok to make them public here. We will not publish an asaned build.

**Comment by jpfeuffer** [comment]
TODO
- disable all options
- fix visibility
- remove forced CXX standard

File: cmake/compiler_flags.cmake [hunk 1]
```diff
@@ -51,8 +51,7 @@ endif()
 if (CMAKE_COMPILER_IS_GNUCXX)
 
   add_compile_options(-Wall -Wextra
-    #-fvisibility=hidden # This is now added as a target property for each library.
-    -Wno-non-virtual-dtor
```
Comment (jpfeuffer) [inline cmake/compiler_flags.cmake:55]: Why did you remove this @timosachsenberg ?
